### PR TITLE
PGP select: better defaults, disclaimers, fixes to protected secret key import

### PR DESCRIPTION
**Pull request by zapu**
Note: it's not necessarily what the ticket asked for. But maybe it could work.  Most importantly, it fixes encrypted key imports :slightly_frowning_face: 

1) `--no-import` is default now. If user wants to import key to KB keyring, they can use `--import` flag.
2) renamed `--only-import` to `--no-publish`. note that `--no-publish` only works with `--import`, otherwise there would be nothing to do.
3) added new gpg ui prompt for when user already has key published, but want to additionally import secret key. user is asked if they want to go ahead and import secret key. Previously this was possible by explicit `pgp select --only-import`, now this case is detected and implicit (but asked by prompt).
To summarise, this improves the following experience:
```
» keybase pgp select
... (key was published to keybase.io but not imported)
# oops, I wanted to import as well.
» keybase pgp select --import --multi
You've already selected this public key for use on Keybase. Would you like to update it on Keybase? [Y/n] y
▶ INFO Key was already up to date.
Do you want to import secret half of this key to local Keybase keyring? [y/N] y
```
4) Added disclaimers where only publishing and importing that explain the process behind the two paths.

### CODE REVIEW

**Comment by zapu** [comment]
@maxtaco can you take a look if this is good enough? It does not follow the ticket exactly, the default prompt for this command (whether you want to import secret or not, if no arguments are provided) would have to be added in engine, so it only asks after user selects key, and I had trouble figuring out set of arguments for it to make sense.

This PR however changes importing by default to not importing by default and adds `--import` option, so maybe it's enough.

It depends how strong do we want push is for local secret key imports. I'll be happy to redo the arguments and add prompt.

File: go/client/cmd_pgp_select.go [hunk 1]
```diff
@@ -31,15 +31,41 @@ func (v *CmdPGPSelect) ParseArgv(ctx *cli.Context) (err error) {
...
+		v.noPublish = ctx.Bool("no-publish")
+		if v.noPublish && !v.importSecret {
+			err = fmt.Errorf("Nothing to do - refused to publish with \"--no-publish\" but did not ask to import secret key (\"--import\")")
 		}
 	}
 	return err
 }
 
+const selectDisclaimer = `You are selecting a PGP key to sign and publish a statement that you
+own said key, effectively making it part of your Keybase.io identity.
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]:
```
You are selecting a PGP key from your local GnuPG keychain, and will
publish a statement signed with this key to make it part of your Keybase.io identity
```

File: go/client/cmd_pgp_select.go [hunk 2]
```diff
@@ -31,15 +31,41 @@ func (v *CmdPGPSelect) ParseArgv(ctx *cli.Context) (err error) {
...
+
+You will be prompted by GPG to unlock your private key - it will be
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]:
```
Note that GnuPG will prompt you to perform this signature.
```

File: go/client/cmd_pgp_select.go [hunk 3]
```diff
@@ -31,15 +31,41 @@ func (v *CmdPGPSelect) ParseArgv(ctx *cli.Context) (err error) {
...
+used to make a signature that you are the owner of said key.
+
+You can also import the secret key to *local*, *encrypted* Keybase
+keyring, to be able to decrypt or sign using Keybase client. To do
+that, use "--import" flag.
+
+Learn more: keybase pgp help select
+
+`
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: is there a chance to bail out here?

File: go/client/cmd_pgp_select.go [hunk 4]
```diff
@@ -31,15 +31,41 @@ func (v *CmdPGPSelect) ParseArgv(ctx *cli.Context) (err error) {
...
+
+const importPrivDisclaimer = `You are selecting a PGP key to publish in your profile, and
+importing secret key to *local*, *encrypted* Keybase keyring.
+
+If your GPG key is encrypted, you will be asked for passphrase
+to unlock it. You may be asked *twice* - first by GPG, to export
+encrypted key bundle, and then by Keybase, to unlock secret key.
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: to unlock *the* secret key.

File: go/client/cmd_pgp_select.go [hunk 5]
```diff
@@ -31,15 +31,41 @@ func (v *CmdPGPSelect) ParseArgv(ctx *cli.Context) (err error) {
...
+
+Please note that this will not work if your secret key lives on a
+hardware device (like smart cards).
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: `like a smart card or a Yubikey`

File: go/client/cmd_pgp_select.go [hunk 6]
```diff
@@ -31,15 +31,41 @@ func (v *CmdPGPSelect) ParseArgv(ctx *cli.Context) (err error) {
...
+that, use "--import" flag.
+
+Learn more: keybase pgp help select
+
+`
+
+const importPrivDisclaimer = `You are selecting a PGP key to publish in your profile, and
+importing secret key to *local*, *encrypted* Keybase keyring.
+
+If your GPG key is encrypted, you will be asked for passphrase
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: be consistent with `GPG` v `GnuPG`.  I don't really care which...

File: go/client/cmd_pgp_select.go [hunk 7]
```diff
@@ -77,26 +112,31 @@ func NewCmdPGPSelect(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Com
...
 			},
 		},
 		Description: `"keybase pgp select" looks at the local GnuPG keychain for all
    available secret keys. It then makes those keys available for use with keybase.
    The steps involved are: (1) sign a signature chain link with the selected PGP
    key and the existing device key; (2) push this signature and the public PGP
-   key to the server; (3) copy the PGP secret half into your local Keybase keyring;
-   and (4) encrypt this secret key with Keybase's local key security
-   mechanism.
+   key to the server; and if "--import" flag is passed: (3) copy the PGP secret half
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: Number: `(1a)`, `(1b)`, `(2a)`, `(2b)`

File: go/client/cmd_pgp_select.go [hunk 8]
```diff
@@ -77,26 +112,31 @@ func NewCmdPGPSelect(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Com
...
+   into your local Keybase keyring; and (4) encrypt this secret key with Keybase's
+   local key security mechanism.
 
    By default, Keybase suggests only one PGP public key, but if you want to,
-   you can supply the "--multi" flag to override this restriction. If you don't
+   you can supply the "--multi" flag to override this restriction. If you
    want your secret key imported into the local Keybase keyring, then use
-   the "--no-import" flag.
+   the "--import" flag. Importing the secret key to Keybase keyring makes
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: "the secret key" -> "your secret key"

File: go/client/cmd_pgp_select.go [hunk 9]
```diff
@@ -77,26 +112,31 @@ func NewCmdPGPSelect(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Com
...
+   it possible to use Keybase PGP like "pgp decrypt" or "pgp sign".
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: "Keybase PGP" -> "Keybase PGP commands"

File: go/engine/errors.go [hunk 10]
```diff
@@ -39,6 +39,6 @@ type PGPImportStubbedError struct {
 }
 
 func (e PGPImportStubbedError) Error() string {
-	return fmt.Sprintf("Key %s has private part stubbed - cannot import to Keybase keychain. Try again with --no-import.",
+	return fmt.Sprintf("Key %s has private part stubbed - cannot import to Keybase keychain. Try again without --import.",
```
Comment (maxtaco) [inline go/engine/errors.go]: `Key %s has a stubbed private key, so we can't import it to the Keybase keychain.`

File: go/client/cmd_pgp_select.go (hunk 3 above)
Comment (zapu) [reply go/client/cmd_pgp_select.go]: Yes, unless key id is explicitly passed in the command line. In default mode of operation this is shown before the key selection prompt where it lists keys and ask you for a number.

**Comment by zapu** [comment]
Text changes done

File: go/client/cmd_pgp_select.go [hunk 11]
```diff
@@ -31,15 +31,42 @@ func (v *CmdPGPSelect) ParseArgv(ctx *cli.Context) (err error) {
...
 	}
 	return err
 }
 
+const selectDisclaimer = `You are selecting a PGP key from your local GnuPG keychain, and
+will publish a statement signed with this key to make it part of 
+your Keybase.io identity.
+
+You will be prompted by GnuPG to unlock your private key - it will
+be used to make a signature that you are the owner of said key.
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: Change the above to: `Note that GnuPG will prompt you to perform this signature.`

File: go/client/cmd_pgp_select.go [hunk 12]
```diff
@@ -31,15 +31,42 @@ func (v *CmdPGPSelect) ParseArgv(ctx *cli.Context) (err error) {
...
+
+You can also import the secret key to *local*, *encrypted* Keybase
+keyring, to be able to decrypt or sign using Keybase client. To do
```
Comment (maxtaco) [inline go/client/cmd_pgp_select.go]: `to be able to decrypt or sign using Keybase Client` -> `enabling decryption and signing with the Keybase client`.

**Comment by zapu** [comment]
Thanks a lot for spending time on this! I'm probably not the right person to write these...

**Comment by maxtaco** [comment]
almost there...

**Comment by zapu** [comment]
small wording changes done
